six/modules/c++/samples/test_six_parsing.cpp

```diff
@@ -111,14 +116,22 @@ void XMLVerifier::verify(const std::string& pathname) const
 
     std::auto_ptr<six::Data> data(six::parseData(mXmlRegistry,
                                                  inStream,
-                                                 six::DataType::COMPLEX,
+                                                 mType,
                                                  mSchemaPaths,
                                                  mLog));
 
     // Write it back out - this verifies both that the XML we write validates
     // against the schema and that our parser writes it without errors
-    six::sicd::ComplexXMLControl xmlControl;
-    std::auto_ptr<xml::lite::Document> xmlDoc(xmlControl.toXML(data.get(),
+    six:: XMLControl* xmlControl;
```

Comment:
Note that this will leak (who deletes `xmlControl`?).  That's why we try to use smart pointers everywhere... see if there's a way you could refactor `six::parseData()` such that you didn't need this check here  and/or if there's a way you can easily do this differently.  I'm trying to avoid the caller needing to specify SICD vs. SIDD on the command line (due to my incredible laziness).  Let me know if you want to catch up on options.